### ISSUE TITLE
[bug report] CHANGE ME TO A SHORT TITLE

### ISSUE THREAD

**Issue by 13584452567** [ookiineko/magiskboot_build#29]
**Basic information**

* Am I using debug build: NO

* OS and machine: Windows 10 22H2 19045.4529

* ZIP file name of the magiskboot binary I am using:  magiskboot-e159716-release-windows-mingw-w64-msvcrt-i686-standalone.zip 

**Description of the bug**

when i tried to unpack ramdisk.cpio，it return a error

**Expected behavior**

extract all files in ramdisk.cpio

**Bug hehavior**

```
magiskboot.exe cpio ./ramdisk.cpio extract
Loading cpio: [ramdisk.cpio]
Extracting entry [acct] to [acct]
Extracting entry [apex] to [apex]
Extracting entry [bin] to [bin]
unknown entry type
Failed to process cpio
```

**Extra links**

[ramdisk.cpio](https://f.ws59.cn/f/edwy8d8at1d)

**Comment by ookiineko** [comment]
offtopic: is that file uploaded by u (?)

**Comment by 13584452567** [comment]
[ramdisk.cpio.zip](https://github.com/user-attachments/files/15882923/ramdisk.cpio.zip)

**Comment by ookiineko** [comment]
thanks for reporting , will be looking into it when i have time

**Comment by ookiineko** [comment]
currently i suspect its a bug in my winsup stub code

**Comment by ookiineko** [comment]
can't produce on linux, issue is windows only

**Comment by 13584452567** [comment]
> can't produce on linux, issue is windows only 

yes,i have tried macos and linux(both x86_64 and arm64).both of them can extract ramdisk.cpio.only windows can't handle this file

**Comment by ookiineko** [comment]
interesting

**Comment by 13584452567** [comment]
> btw maybe try to use the cygwin build [`cygwin-test` (release)](https://github.com/ookiineko/magiskboot_build/releases/tag/cygwin-test) in the meanwhile

well, cygwin-build works fine.it extract all files in cpio.

but this build has too many dll and has no win32 version.i will try to pack all of them into one exe

thanks for your reply!

**Comment by ookiineko** [comment]
i can make a standalone cygwin build, that only depend on cygwin1.dll

**Comment by ookiineko** [comment]
if you want

**Comment by 13584452567** [comment]
> i can make a standalone cygwin build, that only depend on cygwin1.dll

that's really a great news!

i think i need it

thank you very much!~

**Comment by 13584452567** [comment]
a new problem:
command:
magiskboot cpio ramdisk.cpio "extract /system/bin/init init"
output:
```
Loading cpio: [ramdisk.cpio]
Extracting entry [system/bin/init] to [init]
thread '<unnamed>' panicked at 'strerror_r failure', library/std/src/sys/unix/os.rs:127:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
using binary form [this](https://github.com/ookiineko/magiskboot_build/releases/tag/cygwin-test)

**Comment by ookiineko** [comment]
sad
